### DISCUSSION TITLE
How can I test the Dewheater

### DISCUSSION THREAD

**Discussion by Dentonknifeworks** [aaronwmorris/indi-allsky#1424] [Q&A]
I think I have it setup but would like to test it.
I used to use a older dewheater python code that I still use on most of my allskys. I am want to try the built in one on this software.
I got the BME280 setup and working. I think I have the dewheater setup but that is the part I would like to test.
Also what is the best way to install Adafruit-Blinka? I am also new to bookworm and it seems like there is a issue.
Thanks for any help I had to start from scratch today as I had a worn out SD card.

**Comment by aaronwmorris** [comment]
All you should have to do is run the setup.sh script and it will prompt if you want to install the GPIO related python modules.  That is all that needs to be done.

Both Dew Heater modes (standard and PWM) should work well with the Pi4.  The PWM mode does not yet work with Pi5 due to the python modules not working with PWM pins yet (on the Pi5).  The standard mode should work on most SBC systems.

What does your dew heater setup look like?

**Comment by Dentonknifeworks** [comment]
![Screenshot 2024-07-25 at 6 56 04 PM](https://github.com/user-attachments/assets/a6cd3529-1751-4c63-bb2d-ca5d556f801f)
This is the wiring and the relay is on pin 23
bmeAddress 0x76"
 dewHeaterPin 23
To be clear Adafruit-Blainka is installed or satisfied with the GPIO related python modules that I installed in setup?

 
![Screenshot 2024-07-25 at 7 02 22 PM](https://github.com/user-attachments/assets/58683ab5-21f2-4d18-b3ea-de51ec286f5c)
![Screenshot 2024-07-25 at 7 03 16 PM](https://github.com/user-attachments/assets/cc1e4ac6-efc2-44a5-bc78-65b0c177fcee)
I am confused on the D11 is that for pin 11?

**Reply by aaronwmorris** [reply]
If the config saves successfully, then all of the modules should be in place and ready for you.  The config page loads the modules as part of validating the config for the pin numbers provided.  I think you are good to go.  Hopefully, you should be able to hear the relay engage when enabled by the system.

**Reply by Dentonknifeworks** [reply]
Should the sensor_test.py work in the misc dir?
 I am getting no module named 'ephem'

**Reply by Dentonknifeworks** [reply]
Ok I do see the BME280 working in the log. Just want to be sure that I have the correct config for the relay.

**Reply by Dentonknifeworks** [reply]
And now I have the relay clicking when I restart so I think I muddled my way though it....

**Reply by aaronwmorris** [reply]
Excellent!

**Reply by Dentonknifeworks** [reply]
What variable is the temp and humidity stored in from the BME280? I want to display it.

**Reply by aaronwmorris** [reply]
The temperature and humidity will be the first and second variables, respectively, given the assigned variable slot.  If you have the BME280 assigned as Sensor A and Sensor A is assigned `User slot 10`, then temperature will be `{sensor_user_10:0.1f}` and humidity will be `{sensor_user_11:0.1f}`

https://github.com/aaronwmorris/indi-allsky/wiki/Image-Labels#user-sensors

**Reply by Dentonknifeworks** [reply]
Ah got it I had changed the Sensor A Slot that was my issue. This was very confusing to me.
